modules/portaudio/portaudio.c

```diff
@@ -257,18 +258,19 @@ static int play_alloc(struct auplay_st **stp, const struct auplay *ap,
 		      auplay_write_h *wh, void *arg)
 {
 	struct auplay_st *st;
-	PaDeviceIndex dev_index;
+	struct mediadev *dev;
 	int err;
 
 	(void)device;
 
 	if (!stp || !ap || !prm)
 		return EINVAL;
 
-	if (str_isset(device))
-		dev_index = atoi(device);
-	else
-		dev_index = Pa_GetDefaultOutputDevice();
+	dev = !str_cmp(device, "default")
```

Comment:
Should we also allow the empty string for the default device?

Reply:
Looking through the other audio modules, it seems that the idea is indeed that the string can be empty or "default" (matched case-insensitively). In case of errors, we should perhaps return ENODEV instead of ENOENT (cf. coreaudio).